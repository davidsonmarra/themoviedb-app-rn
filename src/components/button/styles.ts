import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

interface Props {
  isDisabled: boolean;
}

const getStyles = ({isDisabled}: Props) =>
  StyleSheet.create({
    container: {
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: 8,
      borderRadius: 12,
      width: '100%',
      opacity: 1,
    },
    primary: {
      backgroundColor: isDisabled ? colors.disabled : colors.primary,
    },
    secondary: {},
  });

export default getStyles;
