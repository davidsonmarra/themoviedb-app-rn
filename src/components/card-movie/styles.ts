import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: colors.borderLight,
    borderRadius: 8,
    marginBottom: 16,
  },
  image: {
    width: 188 / 2,
    height: 282 / 2,
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
  },
  contentDescription: {
    flex: 1,
    padding: 12,
  },
  cardTitle: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.title,
  },
  cardDate: {
    fontSize: 14,
    color: colors.description,
  },
  cardOverview: {
    marginTop: 8,
    fontSize: 14,
    color: colors.text,
  },
});

export default styles;
