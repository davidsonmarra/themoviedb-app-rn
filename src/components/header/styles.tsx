import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 12,
    paddingVertical: 16,
    flexDirection: 'row',
    backgroundColor: colors.primary,
  },
  icon: {
    width: 24,
  },
  text: {
    flex: 1,
    textAlign: 'center',
  },
});

export default styles;
