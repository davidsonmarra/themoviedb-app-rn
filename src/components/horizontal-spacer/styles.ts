import {StyleSheet} from 'react-native';

const getStyles = (width: number) => {
  return StyleSheet.create({
    container: {
      width,
    },
  });
};

export default getStyles;
