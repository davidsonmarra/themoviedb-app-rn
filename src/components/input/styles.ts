import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: colors.border,
    paddingHorizontal: 10,
  },
  input: {
    flex: 1,
    height: 40,
  },
});

export default styles;
