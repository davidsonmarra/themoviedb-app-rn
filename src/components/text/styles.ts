import {StyleSheet} from 'react-native';
import {colors} from '../../configs/theme';

const styles = StyleSheet.create({
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.title,
  },
  text: {
    fontSize: 16,
    fontWeight: 'normal',
    color: colors.text,
  },
  textBold: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.text,
  },
  buttonPrimary: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.buttonText,
    padding: 8,
  },
  buttonSecondary: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.primary,
    padding: 8,
  },
  label: {
    fontSize: 14,
    fontWeight: 'normal',
    color: colors.text,
  },
  error: {
    fontSize: 14,
    fontWeight: 'normal',
    color: colors.error,
  },
  header: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.title,
  },
  overlayTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.buttonText,
  },
  overlayText: {
    fontSize: 16,
    color: colors.buttonText,
  },
});

export default styles;
