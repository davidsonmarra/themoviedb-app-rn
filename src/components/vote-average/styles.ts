import {StyleSheet} from 'react-native';

const getStyles = (color: string) =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      width: 64,
      height: 64,
      borderRadius: 32,
      borderWidth: 5,
      borderColor: color,
    },
    average: {
      color,
      fontSize: 28,
    },
  });

export default getStyles;
