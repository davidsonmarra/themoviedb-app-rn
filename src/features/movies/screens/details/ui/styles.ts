import {StyleSheet} from 'react-native';
import {colors} from '../../../../../configs/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  imageBackground: {
    flex: 1,
  },
  overlay: {
    flex: 1,
    backgroundColor: colors.overlayDark,
    paddingHorizontal: 16,
    paddingVertical: 24,
  },
  title: {
    color: colors.buttonText,
    marginBottom: 8,
    textAlign: 'center',
  },
  containerInfo: {
    flexDirection: 'row',
    marginBottom: 24,
  },
  containerGenres: {
    flexDirection: 'row',
  },
  subtitle: {
    marginBottom: 8,
  },
  average: {
    marginTop: 24,
    marginHorizontal: 'auto',
  },
});

export default styles;
