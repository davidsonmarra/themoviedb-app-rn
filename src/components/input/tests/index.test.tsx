import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Input from '..';
import styles from '../styles';

const renderComponent = (props: any) => render(<Input {...props} />);

describe('Input', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call setValue', () => {
    const mockSetValue = jest.fn();
    renderComponent({
      value: '',
      setValue: mockSetValue,
    });
    const input = screen.getByTestId('input');
    input.props.onChangeText('test');
    expect(mockSetValue).toHaveBeenCalledWith('test');
  });

  it('should set the value on call onChangeText', () => {
    let mockValue = '';
    const mockSetValue = jest
      .fn()
      .mockImplementation(value => (mockValue = value));
    renderComponent({
      value: mockValue,
      setValue: mockSetValue,
    });
    const input = screen.getByTestId('input');
    input.props.onChangeText('test');
    expect(mockValue).toBe('test');
  });

  it('should call onPress', () => {
    const mockOnPress = jest.fn();
    renderComponent({
      onSearch: mockOnPress,
    });
    const button = screen.getByTestId('search-button');
    button.props.onClick();
    expect(mockOnPress).toHaveBeenCalled();
  });
});
