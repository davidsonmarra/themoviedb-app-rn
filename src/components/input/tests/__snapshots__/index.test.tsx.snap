// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#595959",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <TextInput
    style={
      {
        "flex": 1,
        "height": 40,
      }
    }
    testID="input"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="search-button"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="#01B4A4"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={960}
      vbWidth={960}
      width="24"
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278301860,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M784-120L532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56zM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
